shared/teams/team/settings/retention/container.js

```diff
@@ -10,47 +10,98 @@ import {
   withHandlers,
   type TypedState,
 } from '../../../../util/container'
-import {getTeamRetentionPolicy} from '../../../../constants/teams'
+import {getTeamRetentionPolicy, retentionPolicies} from '../../../../constants/teams'
 import {getConversationRetentionPolicy} from '../../../../constants/chat2/meta'
 import {type RetentionPolicy} from '../../../../constants/types/teams'
 import {navigateTo, pathSelector} from '../../../../actions/route-tree'
 import {type Path} from '../../../../route-tree'
 import type {ConversationIDKey} from '../../../../constants/types/chat2'
 import RetentionPicker from './'
 
-// if you supply an onSelect callback, this won't trigger any popup on its own
-// if you don't, the selection will immediately be applied / warning shown on selection
+export type RetentionEntityType = 'adhoc' | 'channel' | 'small team' | 'big team'
+
 export type OwnProps = {
   conversationIDKey?: ConversationIDKey,
-  teamname: string,
-  isTeamWide: boolean,
-  isSmallTeam?: boolean,
+  entityType: RetentionEntityType,
+  teamname?: string,
   type: 'simple' | 'auto',
   onSelect?: (policy: RetentionPolicy, changed: boolean, decreased: boolean) => void,
 }
 
 const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
-  const teamPolicy = getTeamRetentionPolicy(state, ownProps.teamname)
-  const loading = !teamPolicy
-  const policy =
-    !!ownProps.conversationIDKey && getConversationRetentionPolicy(state, ownProps.conversationIDKey)
-  if (!ownProps.isTeamWide && !policy) {
-    throw new Error('Conv retpolicy not present in metaMap')
+  let policy: RetentionPolicy = retentionPolicies.policyRetain
+  let teamPolicy: ?RetentionPolicy
+  let showInheritOption = false
+  let showOverrideNotice = false
+  let loading = false
+
+  switch (ownProps.entityType) {
```

Comment:
I think this code path ends up being a pretty simple mapping and should be fast enough for `mapStateToProps`